test/language/expressions/dynamic-import/usage/nested-arrow-assignment-expression-import-source-returns-thenable.js

```diff
@@ -0,0 +1,32 @@
+// This file was procedurally generated from the following sources:
+// - src/dynamic-import/import-source-returns-thenable.case
+// - src/dynamic-import/default/nested-arrow-assign-expr.template
+/*---
+description: Dynamic import.source() returns a thenable object. (nested arrow)
+esid: sec-import-call-runtime-semantics-evaluation
+features: [source-phase-imports, dynamic-import]
+flags: [generated, async]
+info: |
+    ImportCall :
+        import( AssignmentExpression )
+
+    1. Let referencingScriptOrModule be ! GetActiveScriptOrModule().
+    2. Assert: referencingScriptOrModule is a Script Record or Module Record (i.e. is not null).
+    3. Let argRef be the result of evaluating AssignmentExpression.
+    4. Let specifier be ? GetValue(argRef).
+    5. Let promiseCapability be ! NewPromiseCapability(%Promise%).
+    6. Let specifierString be ToString(specifier).
+    7. IfAbruptRejectPromise(specifierString, promiseCapability).
+    8. Perform ! HostImportModuleDynamically(referencingScriptOrModule, specifierString, promiseCapability).
+    9. Return promiseCapability.[[Promise]].
+
+---*/
+
+let f = () => import.source('./dynamic-import-module_FIXTURE.js').then(imported => {
+
+  assert.sameValue(typeof imported, 'object');
+
+});
+
+
+f().then($DONE, $DONE).catch($DONE);
```

Comment:
The proposal mandates that importing the source of a JS module always throws, so the `assert` is never reached here. There is no way to test that `imported` in an object in test262, and it should instead be done in WPT where there is a known valid module source (i.e. wasm).

Reply:
Updated with negative verifications.